view/elf/elfview.cpp

```diff
@@ -459,15 +459,9 @@ bool ElfView::Init()
 	m_simplifyTemplates = viewSettings->Get<bool>("analysis.types.templateSimplifier", this);
 
 	Ref<Settings> settings = GetLoadSettings(GetTypeName());
-	if (settings && settings->Contains("loader.imageBase") && settings->Contains("loader.architecture")) // handle overrides
-	{
+	if (settings && settings->Contains("loader.imageBase")) // handle overrides
 		preferredImageBase = settings->Get<uint64_t>("loader.imageBase", this);
 
-		Ref<Architecture> arch = Architecture::GetByName(settings->Get<string>("loader.architecture", this));
-		if (!m_arch || (arch && (arch->GetName() != m_arch->GetName())))
-			m_arch = arch;
```

Comment:
m_arch is not being set based on the overridden platform. It is set based on the original platform, which may have changed due to an override.